application/controllers/admin/Tokens.php

```diff
@@ -2365,34 +2365,32 @@ public function kill(int $iSurveyId)
 
         $aData['topBar']['hide'] = true;
 
-        if (!Yii::app()->request->getQuery('ok')) {
+        if (!Yii::app()->request->getPost('ok')) {
             $aData['sidemenu']['state'] = false;
             $aData['backupTableName']   = $newtableDisplay;
-
             $this->renderWrappedTemplate('token', 'deleteParticipantsTable', $aData);
-        } else /* The user has confirmed they want to delete the tokens table */
-        {
-            Yii::app()->db->createCommand()->renameTable("{{{$oldtable}}}", "{{{$newtable}}}");
-
-            $archivedTokenSettings = new ArchivedTableSettings();
-            $archivedTokenSettings->survey_id = $iSurveyId;
-            $archivedTokenSettings->user_id = $userID;
-            $archivedTokenSettings->tbl_name = $newtable;
-            $archivedTokenSettings->tbl_type = 'token';
-            $archivedTokenSettings->created = $DBDate;
-            $archivedTokenSettings->properties = $aData['thissurvey']['tokenencryptionoptions'];
-            $archivedTokenSettings->attributes = json_encode($aData['thissurvey']['attributedescriptions']);
-            $archivedTokenSettings->save();
-
-            //Remove any survey_links to the CPDB
-            SurveyLink::model()->deleteLinksBySurvey($iSurveyId);
+            return;
```

Comment:
Hm shouldn't this be a no-permission error?

Reply:
No : before it's same link with or without &ok=1

Permission is check at https://github.com/LimeSurvey/LimeSurvey/blob/bc26b5b11aa0addae1ecf0cd453eca2ee0685eab/application/controllers/admin/Tokens.php#L1312

If Yii::app()->request->getPost('ok') : we do the actin, if !Yii::app()->request->getPost('ok') we show the confirm action form.

Like before except moving to POST action to have CRSF

Reply:
K